app/models/membership.rb

```diff
@@ -0,0 +1,19 @@
+class Membership < ApplicationRecord
+  belongs_to :user
+
+  validates :joined_at, presence: true
+  validate :single_current_membership
+
+  scope :current, -> { where(left_at: nil) }
+
+  private
+
+  def single_current_membership
+    return if left_at.present?
+
+    existing = self.class.where(left_at: nil, user: user)
```

Comment:
may not need `self.class.` prefix

Reply:
Just double-checked - it is required - definitely would have been neater otherwise 🤷‍♂